garnett/trajectory.py

```diff
@@ -259,12 +280,14 @@ def _validate_input_array(self, value, dim, nelem=None, dtype=None):
 
     def _raw_frame_to_frame(self, raw_frame, dtype=None):
         """Generate a frame object from a raw frame object."""
-        N = len(raw_frame.types)
+        N = len(raw_frame.position)
         ret = FrameData()
 
         mapping = dict()
-        for prop in PARTICLE_PROPERTIES:
-            mapping[prop] = np.asarray(getattr(raw_frame, prop), dtype=dtype)
+        for prop, dtype_ in PARTICLE_PROPERTIES.items():
+            if dtype_ == DEFAULT_DTYPE:
```

Comment:
My solution is less than ideal. We can handle `float32` or `float64` in garnett, and I'm just trying to detect that. I don't think adding a check for `dtype is not None` is helpful because all of the data types are defined at the top of this file. A `None` datatype would concern me here, but I don't think it's possible to have that.

Reply:
I only mentioned it because the `dtype=None` is de default argument for the function. However, now I see that  the function is called only once and a proper argument is always passed. Perhaps removing the default `None` would avoid confusion in the future.